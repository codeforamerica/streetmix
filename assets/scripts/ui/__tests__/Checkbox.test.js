/* eslint-env jest */
import React from 'react'
import { fireEvent, cleanup, render } from '@testing-library/react'
import Checkbox from '../Checkbox'

describe('Checkbox', () => {
  afterEach(cleanup)

  it('renders default snapshot', () => {
    const wrapper = render(<Checkbox>foo</Checkbox>)
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('renders snapshot with all props', () => {
    const wrapper = render(
      <Checkbox
        checked
        disabled
        value="bar"
        id="baz"
        onChange={jest.fn()}
      >
        foo
      </Checkbox>
    )

    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('handles click', () => {
    const handleChange = jest.fn()
    const { getByText } = render(<Checkbox onChange={handleChange}>foo</Checkbox>)
    fireEvent.click(getByText('foo'))
    expect(handleChange).toHaveBeenCalledTimes(1)
  })
})
