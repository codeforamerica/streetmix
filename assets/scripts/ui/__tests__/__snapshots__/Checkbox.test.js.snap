// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders default snapshot 1`] = `
<DocumentFragment>
  <div
    class="checkbox-item"
  >
    <input
      id="checkbox-id-0"
      type="checkbox"
      value=""
    />
    <label
      for="checkbox-id-0"
    >
      foo
    </label>
    <svg
      class="svg-inline--fa fa-check"
    />
  </div>
</DocumentFragment>
`;

exports[`Checkbox renders snapshot with all props 1`] = `
<DocumentFragment>
  <div
    class="checkbox-item"
  >
    <input
      checked=""
      disabled=""
      id="baz"
      type="checkbox"
      value="bar"
    />
    <label
      for="baz"
    >
      foo
    </label>
    <svg
      class="svg-inline--fa fa-check"
    />
  </div>
</DocumentFragment>
`;
